internal/plugin/transport.go

```diff
@@ -70,7 +84,31 @@ func NewTransportHandle(name string, t envelope.Transport) (Handle, error) {
 	if handshake.APIVersion != api.APIVersion {
 		return nil, errHandshakeFailed{
 			Name:   name,
-			Reason: errVersionMismatch{Want: api.APIVersion, Got: handshake.APIVersion},
+			Reason: errAPIVersionMismatch{Want: api.APIVersion, Got: handshake.APIVersion},
```

Comment:
API Version is the Thrift IDL version. The intention is that you can use the handshake to negotiate breaking changes to the API. As long as we don't add new required fields to `HandshakeResponse`, we can make breaking changes to the rest of the IDL and bump the API_VERSION if necessary. API_VERSION will not change every time we make ThriftRW releases.